[DOC] New quick start guide (#1945)
strimzi/strimzi-kafka-operator: documentation/modules/proc-installing-product.adoc

// Module included in the following assemblies:
//
// assembly-evaluation.adoc

[id='proc-install-product-{context}']
= Installing {ProductName}

Using the xref:proc-product-downloads-{context}[download files], install {ProductName} with the Custom Resource Definitions (CRDs) and RBAC configuration required for deployment.

.Prerequisites

> * Installation requires a Kubernetes account with cluster admin credentials

Reviewer: ```suggestion
* Installation requires a Kubernetes account with cluster admin credentials.
```

.Procedure

. Log in to the Kubernetes cluster with cluster admin privileges.

. Modify the installation files to reference a new `kafka` namespace where you will install the {ProductName} Kafka Cluster Operator.
+
NOTE: By default, the files work in the `myproject` namespace.
+
* On Linux, use:
+

[source, shell, subs=+quotes]
----
sed -i 's/namespace: .\*/namespace: kafka/' install/cluster-operator/*RoleBinding*.yaml
----
+
* On Mac, use:
+
[source, shell, subs=+quotes]
----
sed -i '' 's/namespace: .\*/namespace: kafka/' install/cluster-operator/*RoleBinding*.yaml
----

. Create the new `kafka` namespace.
+
[source, shell, subs=+quotes ]
----
kubectl create ns kafka
----

. Deploy the CRDs and role-based access control (RBAC) resources to manage the CRDs.
+
[source, shell, subs=+quotes ]
----
kubectl apply -f install/cluster-operator/ -n kafka
----

> . Create the project `my-kafka-project` where you will deploy your Kafka cluster.

Reviewer: Please remind me, why is it necessary to create this project after creating the `kafka' namespace?

Also, you use the term "project" but the command creates a namespace using the same command as in step 3. This could be confusing for beginners.

+
[source, shell, subs=+quotes ]
----
kubectl create ns my-kafka-project
----

> . Enable the Cluster Operator to watch that namespace.

Reviewer: Please remind me, why is it necessary to run three commands to create the role bindings? I think some further explanation would help beginners.

Author: Added: `The commands create role bindings that grant permission for the Cluster Operator to access the Kafka cluster.`

// Module included in the following assemblies:
//
// assembly-evaluation.adoc

[id='proc-install-product-{context}']
= Installing {ProductName}

Using the xref:proc-product-downloads-{context}[download files], install {ProductName} with the Custom Resource Definitions (CRDs) and RBAC configuration required for deployment.

.Prerequisites

* Installation requires a Kubernetes account with cluster admin credentials.

.Procedure

. Log in to the Kubernetes cluster with cluster admin privileges.

. Modify the installation files to reference a new `kafka` namespace where you will install the {ProductName} Kafka Cluster Operator.
+
NOTE: By default, the files work in the `myproject` namespace.
+
* On Linux, use:
+

[source, shell, subs=+quotes]
----
sed -i 's/namespace: .\*/namespace: kafka/' install/cluster-operator/*RoleBinding*.yaml
----
+
* On Mac, use:
+
[source, shell, subs=+quotes]
----
sed -i '' 's/namespace: .\*/namespace: kafka/' install/cluster-operator/*RoleBinding*.yaml
----

> . Create the new `kafka` namespace for the {ProductName} Kafka Cluster Operator.

Reviewer 2: It still seems backwards to me that we create the namespace only after having told the read about this kafka namespace and had then change the resources. Why can't the namespace creation be the first step?

Author: I guess the idea was to do the prep first. But I've switched.

+
[source, shell, subs=+quotes ]
----
kubectl create ns kafka
----

. Create the `my-kafka-project` namespace where you will deploy your Kafka cluster.
+
[source, shell, subs=+quotes ]
----
kubectl create ns my-kafka-project
----
+

> It is best practice to use namespaces to separate functions.

Reviewer 2: ```suggestion
It is good practice to use namespaces to separate functions.
```

Also, why mention this _after_ all the faffing on with namespaces? Motivate the need for separate namespaces first by saying it's good practice, then create the namespaces then modify the resources seems to me to be the most logical way to do this.

Author: Problem solved. I've moved the note out of the steps to the intro of the procedure: 
In this task you create namespaces in the cluster for your deployment.
It is good practice to use namespaces to separate functions.

. Edit the `install/cluster-operator/050-Deployment-strimzi-cluster-operator.yaml` file and set the `STRIMZI_NAMESPACE` environment variable to the namespace `my-kafka-project`.
+
[source, yaml, subs=+quotes ]
----
        # ...
        env:
        - name: STRIMZI_NAMESPACE
          value: my-kafka-project
        # ...
----

. Deploy the CRDs and role-based access control (RBAC) resources to manage the CRDs.
+
[source, shell, subs=+quotes ]
----
kubectl apply -f install/cluster-operator/ -n kafka
----

> . Enable the Cluster Operator to watch that namespace.

Reviewer 2: ```suggestion
. Give permission to the Cluster Operator (and the Topic Operator it deploys) to watch that namespace
```
